### ISSUE TITLE
Stuck truck and Invisible Truck, cannot scrap or recover

### ISSUE THREAD

**Issue by zuff-coi** [MaFi-Games/Captain-of-Industry-issues#502]
**Describe the issue**
I have 1 stuck truck called "problem1" (north of the settlement next to the
vehicles depot) and 1 invisible pickup which is the last one in the
game. Both can not be scrapped, recovered or moved.
Would it be possible to add button in the vehicle menu to scrap
immediately for the cost of unity and when clicking on a vehicle in the
vehicle overview to not only center the camera on the vehicle but also
open its corresponding menu? That way stuck and invisible vehicles could
be easily removed. This is probably only a quick and dirty method to fix
the corresponding bugs but would give the developers a bit of time to
investigate the problem further and help us, the players, to fix our
problem in the short term.

**Additional context**
 - Game version: 0.4.7a

**Attachments**
[save_22-07-03_19-51-22.zip](https://github.com/MaFi-Games/Captain-of-Industry-issues/files/9041097/save_22-07-03_19-51-22.zip)

**Comment by robbatant** [comment]
i have the same problem with one small excavator stuck which doesnt move or listen to the manual order. at first i could click the recover and scrap button but nothing happend and after that i get a beep if i click these buttons again. says "recovering vehicle (DoJob, Stopped) in the status panel. fuel tank is empty but it doesnt get refueld from pickups. have one fuel stations with 2 pickups assingned and the refueling works for the other working excavators. 
Furthermore i have 4 invisble small excavators which i obviously cant controll. pickups seem to be fine, cant find any invisble ones when i circle through the pickups. 

the small excavator which is stuck is standing at the end of the main bus. for the invisble small excavators you need to circle through them in the vehicles management window. i dont know since which game versions those vehicles are stuck/invisble since i only recently noticed this bug. iam also at game version 0.4.7a

[invisble_stuck excavators.zip](https://github.com/MaFi-Games/Captain-of-Industry-issues/files/9041606/invisble_stuck.excavators.zip)

**Comment by dbiehl1979** [comment]
I too am having the problem.  I have one yellow truck that is sitting in front of my vehicle depot III.  It is one that I tried to recover or scrap (I cannot remember which), and now both buttons are not usable and it just sits there for the past 20 hours of gametime or so.  I also have an excavator (middle size) and a haul dump truck that are invisible and have no access to.  I can add them to mining towers and such, but again nothing happens with them.  I can provide save file if needed.

**Comment by robbatant** [comment]
to update and maybe for info to the update 0.47c, the invisible and stuck excavators reapeard at the vehicle depot. the previously invisible excavators would move to the mining tower after they got fuel and are working fine now. the previously stuck excavator tho is standing infront of the depot and does nothing, see pic1. still (DoJob, Stopped) but with other text.
the weird behaviour of the pickups mentioned under edit in my comment still persist in a different form. bevor this patch they would be stuck at two coal loose storage locations after delivering the coal and at the cargo depot, presumable after delivering fuel to the container ship i presume. now they are all idling empty at a mining tower and i dont know why they went there, see pic2. the selected pickup is moving and will be stuck at the tower after arriving. the picup is numbered because its one of the picups which were also empty idling in the previous version. there are many more pickups with this problem now. weirdly, the mining tower in the picture at which the pickups gather is the one where all the previously invisible excavators are assigned to now.

![pic1](https://user-images.githubusercontent.com/108692666/177641422-4b0a0b7b-9d5d-4bca-9d26-159cbf316c12.png)
![pic2](https://user-images.githubusercontent.com/108692666/177641436-a3c6a202-0c86-48eb-a6b4-d09140bae099.png)

**Comment by robbatant** [comment]
after some testing it seems like some of those idling pickups at the minig tower are following one of the excavators. the pictures pic3 and 4 are 1min apart. looks like the pickups behave like the excavator was a tree harvester?! even tho the pickups are not assigned to the mining tower or anything.

![pic3](https://user-images.githubusercontent.com/108692666/177652362-5cdba880-e1a4-45e8-bcc2-16e75b80dc78.png)
![pic4](https://user-images.githubusercontent.com/108692666/177652370-0472a327-4191-45d3-b3a9-89105458ff47.png)

**Comment by MaFi-Games** [comment]
This should now be fully fixed in v0.4.7d

**Comment by dbiehl1979** [comment]
Yes!  It fixes both of my issues.  The excavator and haul truck that were invisible finally showed back up for work and the yellow truck that was on break for a few hundred years finally decided break time was over.  All trucks are present, accounted for, and back to work.  Thanks for the fixes!

**Comment by zuff-coi** [comment]
> update fixed the issue, well done

Glad to hear! Thanks for your help!
